### PR TITLE
Jono refactor

### PR DESCRIPTION
**Pull request by JonoSommers**
Was able to fix the issue, we needed to initialize @game_statsitics in the SeasonStatistics class.

Everything is working, all tests are passing, and we have a %99.76 test coverage. 

Can everyone review the final product before we merge?

### CODE REVIEW

File: lib/season_statistics.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+class SeasonStatistics
+    attr_reader :teams, :game_teams, :games, :game_statistics
+  
+    def initialize(teams, game_teams, games)
+      @teams = teams
+      @game_teams = game_teams
+      @games = games
+      @game_statistics = GameStatistics.new(@games)
```
Comment (willfox0409) [inline lib/season_statistics.rb:8]: So this was the culprit huh? Wish I could have had the foresight to help you figure this out. Well done sticking with it @JonoSommers .

File: coverage/.last_run.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
   "result": {
-    "line": 99.57
```
Comment (jbleggi) [inline coverage/.last_run.json:3]: Test coverage is looking great.

**Comment by Coralruschak** [comment]
Ditto to Jillian, the coverage is great, thank you!

**Comment by willfox0409** [comment]
@JonoSommers this is a very impressive refactor, much better than what I was experimenting with last night. So glad we figured out how to get all the tests to keep passing too. It's significantly more concise and organized now that the methods are parsed out into new classes and not piling up on each-other in the StatTracker! There's also way less repeated code, so a truly DRY fix, satisfying to see this design principle put to use. Fantastic job!

**Pull request merged**